### PR TITLE
PMM-6091 Rewrite pmm update panel to reactjs

### PR DESCRIPTION
**Pull request by nicolalamacchia**
[PMM-6091](https://jira.percona.com/browse/PMM-6091)

- [ ] https://github.com/Percona-Lab/pmm-submodules/pull/938

### CODE REVIEW

**Comment by it-percona** [comment]
[CLA assistant check](https://cla-assistant.percona.com:443/percona/grafana-dashboards?pullRequest=766) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.percona.com:443/percona/grafana-dashboards?pullRequest=766) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.percona.com:443/check/percona/grafana-dashboards?pullRequest=766) it.</sub>

File: pmm-app/src/pmm-update-panel/components/CenteredButton/CenteredButton.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { Button, ButtonProps } from '@grafana/ui';
+
+import * as styles from './CenteredButton.styles';
+
+export const CenteredButton = ({ children, className, ...props }: ButtonProps) => (
+  <Button className={`${className} ${styles.centeredButton}`} {...props}>
```
Comment (atymchuk) [inline pmm-app/src/pmm-update-panel/components/CenteredButton/CenteredButton.tsx:7]: What if I wand to pass className as undefined? Pls check out `import { cx } from 'emotion';`. It's properly handle undefined )

File: pmm-app/src/pmm-update-panel/components/LastCheck/LastCheck.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import * as styles from './LastCheck.styles';
+
+interface LastCheckProps {
+  onCheckForUpdates: (e: MouseEvent) => void;
+  isLoading: boolean;
+  lastCheckDate: string;
+}
+
+export const LastCheck = ({ isLoading, lastCheckDate, onCheckForUpdates }: LastCheckProps) => {
+  const icon = `fa ${isLoading ? 'fa-spin ' : ''}fa-refresh`;
```
Comment (atymchuk) [inline pmm-app/src/pmm-update-panel/components/LastCheck/LastCheck.tsx:13]: Grafana has its own spinner, we could use them here. The plan is to get rid fontawesome :)

File: pmm-app/src/pmm-update-panel/components/UpdateButton/UpdateButton.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import * as styles from './UpdateButton.styles';
+
+interface UpdateButtonProps {
+  onClick: () => void;
+  nextVersion: string;
+  disabled?: boolean;
+}
+
+export const UpdateButton = ({ onClick, nextVersion, disabled = false }: UpdateButtonProps) => (
+  <CenteredButton disabled={disabled} className={styles.updateButton} onClick={onClick} icon="fa fa-download">
```
Comment (atymchuk) [inline pmm-app/src/pmm-update-panel/components/UpdateButton/UpdateButton.tsx:13]: Pls try to use grafana icons from '@grafana/ui' vs fontawesome.

File: pmm-app/src/pmm-update-panel/plugin.json [hunk 4]
```diff
@@ -1,5 +1,30 @@
...
+    "description": "A panel to visualize and install updates",
+    "author": {
+      "name": "Percona",
+      "url": "https://percona.com"
+    },
+    "keywords": ["percona", "plugin", "updates"],
+    "links": [
+      {
+        "name": "Website",
+        "url": "https://github.com/percona/grafana-dashboards/pmm-app/src/pmm-update-panel"
```
Comment (atymchuk) [inline pmm-app/src/pmm-update-panel/plugin.json:15]: Don't forget to rename to `pmm-app/src/pmm-update` :)

File: pmm-app/src/pmm-update-panel/components/CenteredButton/CenteredButton.tsx (hunk 1 above)
Comment (nicolalamacchia) [reply pmm-app/src/pmm-update-panel/components/CenteredButton/CenteredButton.tsx:7]: Neat! Thanks, done

File: pmm-app/src/pmm-update-panel/components/LastCheck/LastCheck.tsx (hunk 2 above)
Comment (nicolalamacchia) [reply pmm-app/src/pmm-update-panel/components/LastCheck/LastCheck.tsx]: Grafana's documentation says that the `Icon` component is a wrapper around [Font Awesome and Unicons icons](https://developers.grafana.com/ui/latest/index.html?path=/docs/docs-overview-icon--icons-overview) (and I guess that applies also for the `Button` component and so on)

File: pmm-app/src/pmm-update-panel/components/UpdateButton/UpdateButton.tsx (hunk 3 above)
Comment (nicolalamacchia) [reply pmm-app/src/pmm-update-panel/components/UpdateButton/UpdateButton.tsx]: Same as [this one](https://github.com/percona/grafana-dashboards/pull/766/files/4def364973bdb9834088795560da738f6d7fca77#r440811552)

File: pmm-app/src/pmm-update-panel/components/LastCheck/LastCheck.tsx (hunk 2 above)
Comment (atymchuk) [reply pmm-app/src/pmm-update-panel/components/LastCheck/LastCheck.tsx]: That's not the point. The point is sustainability, which we won't achieve if we use classes like 'fa-spin'.

**Comment by atymchuk** [comment]
@nicolalamacchia Any reason the tests fail?

**Comment by nicolalamacchia** [comment]
> I would appreciate it if we can squash ts-ignore )

Done

**Pull request merged**